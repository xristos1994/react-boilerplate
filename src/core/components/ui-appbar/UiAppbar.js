import React from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import AccountCircle from "@material-ui/icons/AccountCircle";
import MenuItem from "@material-ui/core/MenuItem";
import Menu from "@material-ui/core/Menu";

import { config } from "@core/configuration";
import { isLogged, coreAuth_logout } from "@core/models/authentication/props";
import {
  coreUi_openDrawer,
  coreUi_closeDrawer,
  drawerProps
} from "@core/models/core-ui/props";
import { withProps } from "@core/utils/props";
import { styles } from "./style";

export const UiAppbar = ({
  isLogged,
  coreAuth_logout,
  coreUi_openDrawer,
  coreUi_closeDrawer,
  drawerProps
}) => {
  const classes = styles();

  const [anchorEl, setAnchorEl] = React.useState(null);
  const open = Boolean(anchorEl);

  const handleMenu = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          {isLogged && (
            <IconButton
              edge="start"
              className={classes.menuButton}
              color="inherit"
              aria-label="menu"
              onClick={() =>
                drawerProps.show ? coreUi_closeDrawer() : coreUi_openDrawer()
              }
            >
              <MenuIcon />
            </IconButton>
          )}
          <Typography variant="h6" className={classes.title}>
            React Boilerplate
          </Typography>
          {isLogged && config.hasLogin && (
            <div>
              <IconButton
                aria-label="account of current user"
                aria-controls="menu-appbar"
                aria-haspopup="true"
                onClick={handleMenu}
                color="inherit"
              >
                <AccountCircle />
              </IconButton>
              <Menu
                id="menu-appbar"
                anchorEl={anchorEl}
                anchorOrigin={{
                  vertical: "top",
                  horizontal: "right"
                }}
                keepMounted
                transformOrigin={{
                  vertical: "top",
                  horizontal: "right"
                }}
                open={open}
                onClose={handleClose}
              >
                <MenuItem
                  onClick={() => {
                    handleClose();
                    coreAuth_logout();
                  }}
                >
                  Log out
                </MenuItem>
              </Menu>
            </div>
          )}
        </Toolbar>
      </AppBar>
    </div>
  );
};
export default withProps({
  isLogged,
  coreAuth_logout,
  coreUi_openDrawer,
  coreUi_closeDrawer,
  drawerProps
})(UiAppbar);
