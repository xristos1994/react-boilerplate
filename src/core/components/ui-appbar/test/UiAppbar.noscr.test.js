import React from 'react';
import { shallow } from 'enzyme';
import { UiAppbar } from './../UiAppbar';

describe('Appbar Test', () => {
  it('Renders Appbar for Logged In', () => {
    const props = {
      isLogged: true,
      drawerProps: { show: true },
    };

    const component = shallow(<UiAppbar {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('Renders Appbar for not Logged In', () => {
    const props = {
      isLogged: false,
      drawerProps: { show: true },
    };

    const component = shallow(<UiAppbar {...props} />);
    expect(component).toMatchSnapshot();
  });
});
