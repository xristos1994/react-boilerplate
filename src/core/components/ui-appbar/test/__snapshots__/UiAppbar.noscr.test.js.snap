// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appbar Test Renders Appbar for Logged In 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="menu"
        className="makeStyles-menuButton-2"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        variant="h6"
      >
        React Boilerplate
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;

exports[`Appbar Test Renders Appbar for not Logged In 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        variant="h6"
      >
        React Boilerplate
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
